### PR TITLE
Implement Item TTL for InMemory store

### PR DESCRIPTION
**Pull request by joe94**

### CODE REVIEW

File: store/inmem/InMem.go [hunk 1]
```diff
@@ -56,17 +70,23 @@ func (i *InMem) Get(key model.Key) (store.OwnableItem, error) {
...
 func (i *InMem) GetAll(bucket string) (map[string]store.OwnableItem, error) {
 	i.lock.RLock()
 	defer i.lock.RUnlock()
 	items := i.data[bucket]
 	if items == nil {
-		items = map[string]store.OwnableItem{}
+		items = map[string]expireableItem{}
 	}
-	return items, nil
+	i.pruneExpiredItems(bucket, items)
```
Comment (kristinapathak) [inline store/inmem/InMem.go:88]: Since we only have a read lock, I don't think we can prune here?

File: store/inmem/InMem.go [hunk 2]
```diff
@@ -56,17 +70,23 @@ func (i *InMem) Get(key model.Key) (store.OwnableItem, error) {
 	if !ok {
 		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "get"})
 	}
-	return item, nil
+	if i.itemExpired(item) {
+		i.pruneExpiredItem(key.Bucket, key.ID, bucket)
```
Comment (kristinapathak) [inline store/inmem/InMem.go:74]: Since we only have a read lock, I don't think we can prune here?

File: store/inmem/InMem.go [hunk 3]
```diff
@@ -80,6 +100,48 @@ func (i *InMem) Delete(key model.Key) (store.OwnableItem, error) {
...
+	}
 	delete(bucket, key.ID)
-	return item, nil
+	if len(bucket) == 0 {
+		delete(i.data, key.Bucket)
+	}
+	return item.OwnableItem, nil
+}
+
+func (i *InMem) pruneExpiredItem(bucketName string, itemID string, bucket map[string]expireableItem) {
```
Comment (kristinapathak) [inline store/inmem/InMem.go:115]: Since we don't check that this is expired, I suggest renaming to `pruneItem` (or moving the expired check here).

**Review by kristinapathak — COMMENTED** [review]
Overall looks good! Have a few lock concerns that I left comments for.

File: store/inmem/InMem.go (hunk 1 above)
Comment (joe94) [reply store/inmem/InMem.go:88]: Good catch! Not sure how the tests here were passing when the first time I ran `go test -race` failed immediately locally. I added a fix and also some documentation explicitly stating this is not recommended for production environments.

**Review by kristinapathak — APPROVED** [review]
💯

**Pull request merged**